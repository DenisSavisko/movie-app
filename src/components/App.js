import React from 'react';
import { Provider } from 'react-redux';
import configureStore from '../store/configureStore';
import { Router } from "react-router-dom";
import history from '../history';

import Header from './Header';
import Main from './Main';
import Footer from './Footer';

const store = configureStore(); // You can also pass in an initialState here

history.listen(({pathname}) => { // location to redux;
  if(store.getState().state.urlPathNow !== pathname){
    store.dispatch({type:'PAGE_CHANGE', payload:1}); // remove state.json only when path changed
    store.dispatch({type:'FETCH_TO_STATE', payload:{}}); // remove state.json only when path changed
    store.dispatch({type:'HISTORY_UPDATE', payload:pathname});
  }
});

export default () =>
  <Provider store={store}>
    <Router history={history}>
      <Header />
      <Main />
      <Footer />
    </Router>
  </Provider>