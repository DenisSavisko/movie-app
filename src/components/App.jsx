import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import configureStore from '../store/configureStore';
import history from '../history';

import Header from './Header';
import Main from './Main';
import Footer from './Footer';

export const store = configureStore(); // You can also pass in an initialState here
export default () => (
  <Provider store={store}>
    <Router history={history}>
      <div className="content">
        <Header />
        <Main />
      </div>
      <Footer />
    </Router>
  </Provider>
);
