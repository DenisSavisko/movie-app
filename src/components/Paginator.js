import React from 'react';
import { Container, Button, Row, Col }  from 'react-bootstrap'


export default ({page, totalPages, handlePageChange})=>
<Container className='text-center'>
  <Row>
    <Col>

      { 
        page > 1 && 
        <Button variant="outline-primary" value='prev' onClick={handlePageChange}>
          {'<-Prev'}
        </Button>

      }
      {page && 
      <Button variant="outline-primary" disabled>
          {page}
        </Button>
      }
      { 
        page < totalPages && 
        <Button variant="outline-primary" value='next' onClick={handlePageChange}>
          {'Next->'}
        </Button>
      }
    </Col>
  </Row>
</Container>