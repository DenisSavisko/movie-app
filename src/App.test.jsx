import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from './components/App';
import Paginator from './components/Paginator';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Paginator container', () => {
  const props = {
    page: 2,
    totalPages: 5,
    handlePageChange: () => true,
  };

  test('renders properly', () => {
    const wrapper = shallow(<Paginator {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('paginator has "<-Prev" btn', () => {
    const wrapper = shallow(<Paginator {...props} />);
    expect(
      wrapper
        .find('Button')
        .first()
        .text()
    ).toEqual('<-Prev');
  });

  test('paginator buttons quantity', () => {
    const nextProps = {
      ...props,
      page: 3,
    };
    const wrapper = shallow(<Paginator {...nextProps} />);
    expect(wrapper.find('Button').length).toBe(3);
  });

  test('onclick dispatches "handlePageChange", it\'s gotten from props', () => {
    const mockHandlePageChange = jest.fn();
    const newProps = {
      ...props,
      handlePageChange: mockHandlePageChange,
    };
    const wrapper = shallow(<Paginator {...newProps} />);
    wrapper
      .find('Button')
      .first()
      .simulate('click');
    expect(mockHandlePageChange).toHaveBeenCalledTimes(1);
  });
});
