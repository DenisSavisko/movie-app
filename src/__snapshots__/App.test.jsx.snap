// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator container renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    handlePageChange={[Function]}
    page={2}
    totalPages={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": <ForwardRef(render)>
        <ForwardRef
          as="div"
        >
          <ForwardRef
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            value="prev"
            variant="outline-secondary"
          >
            &lt;-Prev
          </ForwardRef>
          <ForwardRef
            active={false}
            disabled={true}
            type="button"
            variant="outline-secondary"
          >
            2
          </ForwardRef>
          <ForwardRef
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            value="next"
            variant="outline-secondary"
          >
            Next-&gt;
          </ForwardRef>
        </ForwardRef>
      </ForwardRef(render)>,
      "className": "text-center my-5",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          as="div"
        >
          <ForwardRef
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            value="prev"
            variant="outline-secondary"
          >
            &lt;-Prev
          </ForwardRef>
          <ForwardRef
            active={false}
            disabled={true}
            type="button"
            variant="outline-secondary"
          >
            2
          </ForwardRef>
          <ForwardRef
            active={false}
            disabled={false}
            onClick={[Function]}
            type="button"
            value="next"
            variant="outline-secondary"
          >
            Next-&gt;
          </ForwardRef>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": Array [
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="prev"
              variant="outline-secondary"
            >
              &lt;-Prev
            </ForwardRef>,
            <ForwardRef
              active={false}
              disabled={true}
              type="button"
              variant="outline-secondary"
            >
              2
            </ForwardRef>,
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="next"
              variant="outline-secondary"
            >
              Next-&gt;
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "<-Prev",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "value": "prev",
              "variant": "outline-secondary",
            },
            "ref": null,
            "rendered": "<-Prev",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": 2,
              "disabled": true,
              "type": "button",
              "variant": "outline-secondary",
            },
            "ref": null,
            "rendered": 2,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": false,
              "children": "Next->",
              "disabled": false,
              "onClick": [Function],
              "type": "button",
              "value": "next",
              "variant": "outline-secondary",
            },
            "ref": null,
            "rendered": "Next->",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "active": false,
                "disabled": false,
                "type": "button",
                "variant": "primary",
              },
              "displayName": "Button",
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": <ForwardRef(render)>
          <ForwardRef
            as="div"
          >
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="prev"
              variant="outline-secondary"
            >
              &lt;-Prev
            </ForwardRef>
            <ForwardRef
              active={false}
              disabled={true}
              type="button"
              variant="outline-secondary"
            >
              2
            </ForwardRef>
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="next"
              variant="outline-secondary"
            >
              Next-&gt;
            </ForwardRef>
          </ForwardRef>
        </ForwardRef(render)>,
        "className": "text-center my-5",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
          >
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="prev"
              variant="outline-secondary"
            >
              &lt;-Prev
            </ForwardRef>
            <ForwardRef
              active={false}
              disabled={true}
              type="button"
              variant="outline-secondary"
            >
              2
            </ForwardRef>
            <ForwardRef
              active={false}
              disabled={false}
              onClick={[Function]}
              type="button"
              value="next"
              variant="outline-secondary"
            >
              Next-&gt;
            </ForwardRef>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": Array [
              <ForwardRef
                active={false}
                disabled={false}
                onClick={[Function]}
                type="button"
                value="prev"
                variant="outline-secondary"
              >
                &lt;-Prev
              </ForwardRef>,
              <ForwardRef
                active={false}
                disabled={true}
                type="button"
                variant="outline-secondary"
              >
                2
              </ForwardRef>,
              <ForwardRef
                active={false}
                disabled={false}
                onClick={[Function]}
                type="button"
                value="next"
                variant="outline-secondary"
              >
                Next-&gt;
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "children": "<-Prev",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
                "value": "prev",
                "variant": "outline-secondary",
              },
              "ref": null,
              "rendered": "<-Prev",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "children": 2,
                "disabled": true,
                "type": "button",
                "variant": "outline-secondary",
              },
              "ref": null,
              "rendered": 2,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "children": "Next->",
                "disabled": false,
                "onClick": [Function],
                "type": "button",
                "value": "next",
                "variant": "outline-secondary",
              },
              "ref": null,
              "rendered": "Next->",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "active": false,
                  "disabled": false,
                  "type": "button",
                  "variant": "primary",
                },
                "displayName": "Button",
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
